libraries/Microsoft.Bot.Builder.Ai/Translator.cs

```diff
@@ -82,15 +82,19 @@ public void AddNoTranslatePhrase(string noTranslatePhrase)
                     string[] wordIndexes = alignData.Split('-');
                     int srcStartIndex = Int32.Parse(wordIndexes[0].Split(':')[0]);
                     int srcLength = Int32.Parse(wordIndexes[0].Split(':')[1]) - srcStartIndex + 1;
+                    if ((srcLength + srcStartIndex) > sourceMessage.Length)
```

Comment:
Comments here around intent would be good. I have *No IDEA* what this code is trying to do.

Reply:
this code is trying to avoid exceptions thrown in case of wrong alignment information sent by the translator